lms/djangoapps/ccx/migrations/0004_seed_forum_roles_in_ccx_courses.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from ccx_keys.locator import CCXLocator
+from courseware.courses import get_course_by_id
+from django.db import migrations
+from django.http import Http404
+
+from django_comment_common.models import FORUM_ROLE_ADMINISTRATOR, FORUM_ROLE_MODERATOR, \
+    FORUM_ROLE_COMMUNITY_TA, FORUM_ROLE_STUDENT
+from django_comment_common.utils import STUDENT_ROLE_PERMISSIONS,  MODERATOR_ROLE_PERMISSIONS, \
+    ADMINISTRATOR_ROLE_PERMISSIONS
+
+
+def seed_forum_roles_for_existing_ccx(apps, schema_editor):
+    """
+    Seed forum roles and make CCX coach forum admin for respective CCX(s).
+
+    Arguments:
+        apps (Applications): Apps in edX platform.
+        schema_editor (SchemaEditor): For editing database schema i.e create, delete field (column)
+
+    """
+    CustomCourseForEdX = apps.get_model("ccx", "CustomCourseForEdX")
+    Role = apps.get_model("django_comment_common", "Role")
+    Permission = apps.get_model("django_comment_common", "Permission")
+    db_alias = schema_editor.connection.alias
+    # This will need to be changed if ccx gets moved out of the default db for some reason.
+    if db_alias != 'default':
+        return
+
+    for ccx in CustomCourseForEdX.objects.using(db_alias).all():
+        if not ccx.course_id or ccx.course_id.deprecated:
+            # prevent migration for deprecated course ids or invalid ids.
+            continue
+
+        ccx_locator = CCXLocator.from_course_locator(ccx.course_id, unicode(ccx.id))
+
+        # Create forum roles.
+        role_query_set = Role.objects.using(db_alias)
+        admin_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_ADMINISTRATOR, course_id=ccx_locator)
+        moderator_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_MODERATOR, course_id=ccx_locator)
+        community_ta_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_COMMUNITY_TA, course_id=ccx_locator)
+        student_role, _ = role_query_set.get_or_create(name=FORUM_ROLE_STUDENT, course_id=ccx_locator)
+
+        # Add permissions for each role.
+        permission_query_set = Permission.objects.using(db_alias)
+        for name in ADMINISTRATOR_ROLE_PERMISSIONS:
+            admin_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for name in MODERATOR_ROLE_PERMISSIONS:
+            moderator_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for name in STUDENT_ROLE_PERMISSIONS:
+            student_role.permissions.add(permission_query_set.get_or_create(name=name)[0])
+        for permission in student_role.permissions.all():
+            moderator_role.permissions.add(permission)
+        for permission in moderator_role.permissions.all():
+            community_ta_role.permissions.add(permission)
+        for permission in moderator_role.permissions.all():
+            admin_role.permissions.add(permission)
```

Comment:
@ormsbee 

> What happens if this gets partway through and errors for some reason? Is it possible to re-run the command? Will that leave extra data rows in the system?

No, it uses `get_or_create` to create new roles and permissions, so it won't create extra rows if they already exist. `role.permissions.add(permission)` also won't create a new row if role already has the permission.